parsers/pascal.c

```diff
@@ -74,6 +80,68 @@ static bool tail (const char *cp)
 	return result;
 }
 
+static void parseArglist(const char *buf, char **arglist, char **vartype)
```

Comment:
Using vString() for storing arglist and vartype is an alternative approach.

In a micro view, an operation on vString tasks higher const than using `char *` directly.
However, in a macro view, vString is not so bad. Using char *, you must call strdup twice in every calling parseArglist.
vString can be reused once it is new'ed in findPascalTags. 
A vString can clear its internal buffer keeping char * with vStringClear.
Though, it is cleared, the buffer is not deallocated. 

Writing English is hard for me. Let's show me in pseudo-code:
```
/* Your code using raw C string */

static void parseArglist(const char *buf, char **arglist, char **vartype)
{
  *arglist = strdup();
  *vartype = strdup();  
}

static void findPascalTags (void)
{
  char *arglist = NULL;
  char *vartype = NULL;
  // ...
  loop() {
    createPascalTag();
    // ...
    free (arglist);
    free (vartype);    
    parseArglist(, &arglist, &vartype);
  }
  free (arglist);
  free (vartype);    
  
}

/* What we can do with vString */
static void parseArglist(const char *buf, vString *arglist, vString *vartype)
{
  char c, *str;
  /* ... */
  vStringPut(arglist, c);
  /* or */ vStringCatS(arglist, str);
  /* or */ vStringNCatS(arglist, str, len);
  vStringPut(vartype, c);
  /* or */ vStringCatS(vartype, str);
  /* or */ vStringNCatS(vartype, str, len);

}

static void findPascalTags (void)
{
  vString *arglist = vStringNew();
  vString *vartype = vStringNew();
  // ...
  loop() {
    createPascalTag();
    // ...
    vStringClear (arglist);
    vStringClear (vartype);    
    parseArglist(, arglist, vartype);
  }
  vStringDelete (arglist);
  vStringDelete (vartype);    
  
}
```

Of course, I don't care about the performance of this level.
However, vString helps me understand the logic.

This comment is not for "must-fix". Your code is already works fine. So this is a just comment.

Reply:
Great idea and thanks for the detailed explanation.
I rewrote the code to use `vString` (back in 2009 when I wrote the code initially I wasn't so much aware of the `vString` helper and didn't touch the since then).

It's simpler now and we don't need any more string copying than what is done implicitly by `vStringCat*()`.